diff --git a/BUILD b/BUILD
index 072317e..e86f4d4 100644
--- a/BUILD
+++ b/BUILD
@@ -12,6 +12,6 @@ config_setting(

 load(":bazel/gflags.bzl", "gflags_sources", "gflags_library")

-(hdrs, srcs) = gflags_sources(namespace=["gflags", "google"])
+(hdrs, srcs) = gflags_sources(namespace=["google", "gflags"])
 gflags_library(hdrs=hdrs, srcs=srcs, threads=0)
 gflags_library(hdrs=hdrs, srcs=srcs, threads=1)
